### DISCUSSION TITLE
Youtube Music Search API

### DISCUSSION THREAD

**Discussion by Zintom** [Tyrrrz/YoutubeExplode#750] [General]
Is support for the Youtube Music search API being considered? I'd like to easily filter out music videos etc and cut down to just the high quality pure music uploads used by Music.

**Comment by Tyrrrz** [comment]
Not sure how complex that it is. If it's as simple as adding a new filter here, then it should be fine:

https://github.com/Tyrrrz/YoutubeExplode/blob/59d642788aa03967a791806b29a55e55a9b3c510/YoutubeExplode/Search/SearchController.cs#L30-L35

**Reply by Zintom** [reply]
I don't think a simple search filter will work as YouTube itself doesn't contain a search filter for music only.
You know a lot more about the backend than me; from what I can see YouTube will upload a "sound-only" auto-generated track, usually on some random channel with the name of the artist (not the official channel).

The music ID is the same as the video ID. For example if you listen to https://music.youtube.com/watch?v=GJvGf_ifiKw, you can just change the start of the URL to "www." and it will take you to the auto-generated video.

**Reply by Zintom** [reply]
> Not sure how complex that it is. If it's as simple as adding a new filter here, then it should be fine:
> 
> https://github.com/Tyrrrz/YoutubeExplode/blob/59d642788aa03967a791806b29a55e55a9b3c510/YoutubeExplode/Search/SearchController.cs#L30-L35

How does that search filter actually work?

**Reply by Tyrrrz** [reply]
It just sends the same parameters that your browser does when you filter search results

**Reply by Zintom** [reply]
I'll investigate then to see if any extra params are sent.
